spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/instrument/web/HttpTraceKeysInjectorUnitTests.java

```diff
@@ -0,0 +1,67 @@
+package org.springframework.cloud.sleuth.instrument.web;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+import java.net.URL;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Random;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.cloud.sleuth.DefaultSpanNamer;
+import org.springframework.cloud.sleuth.NoOpSpanReporter;
+import org.springframework.cloud.sleuth.Span;
+import org.springframework.cloud.sleuth.SpanNamer;
+import org.springframework.cloud.sleuth.SpanReporter;
+import org.springframework.cloud.sleuth.TraceKeys;
+import org.springframework.cloud.sleuth.Tracer;
+import org.springframework.cloud.sleuth.log.Slf4jSpanLogger;
+import org.springframework.cloud.sleuth.log.SpanLogger;
+import org.springframework.cloud.sleuth.sampler.AlwaysSampler;
+import org.springframework.cloud.sleuth.trace.DefaultTracer;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+
+
+public class HttpTraceKeysInjectorUnitTests {
+	TraceKeys traceKeys = new TraceKeys();
+	private SpanNamer spanNamer = new DefaultSpanNamer();
+	private SpanLogger spanLogger = new Slf4jSpanLogger("skip");
+	private SpanReporter spanReporter = new NoOpSpanReporter();
+	AlwaysSampler sampler = new AlwaysSampler();
+	Tracer tracer = new DefaultTracer(sampler, new Random(), spanNamer, spanLogger, spanReporter, traceKeys);
+	HttpTraceKeysInjector injector;
+	
+	@Before
+	public void setup() {
+		assertNotNull(tracer);
+		injector = new HttpTraceKeysInjector(tracer, traceKeys);
+	}
+	
+	@Test
+	public void testHttpAttributes() throws Exception {
```

Comment:
this test say nothing - I still have no idea what you are trying to achieve even though I asked twice already this question AFAIR